coalib/bears/Bear.py

```diff
@@ -34,6 +38,21 @@ def do_quit(self, arg):
     do_q = do_quit
     do_exit = do_quit
 
+    def do_settings(self, arg):
+        if self.bear is not None:
+            md = self.bear.get_metadata()
+            section_params_dict = md.create_params_from_section(
+                                  self.bear.section)
+            for param in md.non_optional_params:
+                self.message('%s = %r' % (param, section_params_dict[param]))
+            for param in md.optional_params:
+                self.message('%s = %r' % (param, section_params_dict[param] if
+                                          param in section_params_dict else
+                                          md.optional_params[param][2]))
+            return 1
+        else:
+            self.error("Name 'settings' is not defined.")
```

Comment:
when there bear object is none i.e., `debug_run(self.run, Debugger(bear=None), *args, **kwargs)` then if use tries to use the settings commands it will give a message `"Name 'settings' is not defined."` (self.error is actually `print` function so it will not interrupt the whole program)

Reply:
If not passing a bear is an erroneous state, raise an exception while instantiating the debugger. This is easiest and will always require you to initialize the debugger properly.